### PR TITLE
Export the metadata team -> groupBy in metadata-uploader

### PR DESCRIPTION
**Pull request by yuli-han**
## Summary

Add metadata export the for team -> chronon entity(group_by and join) in metadata-uploader.
The Mussel data name CHRONON_ENTITY_BY_TEAM has been added by mussel team. https://airbnb.slack.com/archives/C01SS67TEUQ/p1718730788359319

group_bys/trust_v21 -> sample_group_by.v1, 

## Test Plan

- [ ] Added Unit Tests
- [ ] Covered by existing CI
- [x] Integration tested
Local tests for groupBy: 
python3 ~/.local/bin/run.py --mode=metadata-upload  --conf production/group_bys// --chronon-jar ~/test/chronon-embedded.jar | tee ~/test/metadata_upload.log

Sample log:

```
2024-06-24 20:54:22 INFO  MetadataStore:204 - Putting metadata for
dataset: CHRONON_ENTITY_BY_TEAM
key: group_bys/cs_ds
conf: List(group_bys/cs_ds/test.v1)
```

Local tests for joins:
python3 ~/.local/bin/run.py --mode=metadata-upload  --conf production/joins// --chronon-jar ~/test/chronon-embedded.jar | tee ~/test/metadata_upload.log

sample log:
```
2024-06-24 20:56:14 INFO  MetadataStore:204 - Putting metadata for
dataset: CHRONON_ENTITY_BY_TEAM
key: joins/knowledge_graph
conf: List(joins/knowledge_graph/test.v1)
```

Local test fetching:

2024-06-24 22:44:20 INFO  Fetcher:206 - [test fetcher] Success(ArraySeq(group_bys/trust_test/test_v1))

## Checklist
- [ ] Documentation update

## Reviewers

@haozhen-ding

### CODE REVIEW

File: online/src/main/scala/ai/chronon/online/MetadataEndPoint.scala [hunk 1]
```diff
@@ -15,7 +15,7 @@ object MetadataEndPoint {
   @transient implicit lazy val logger = LoggerFactory.getLogger(getClass)
 
   val ConfByKeyEndPointName = "ZIPLINE_METADATA"
-  val NameByTeamEndPointName = "CHRONON_METADATA_BY_TEAM"
+  val NameByTeamEndPointName = "CHRONON_ENTITY_BY_TEAM"
```
Comment (pengyu-hou) [inline online/src/main/scala/ai/chronon/online/MetadataEndPoint.scala:18]: do we have to hard code it ?

File: online/src/main/scala/ai/chronon/online/MetadataStore.scala [hunk 2]
```diff
@@ -68,6 +69,51 @@ class MetadataStore(kvStore: KVStore, val dataset: String = ChrononMetadataKey,
       }
   }
 
+  def getEntityListByTeam[T <: TBase[_, _]: Manifest](team: String): Try[Seq[String]] = {
```
Comment (pengyu-hou) [inline online/src/main/scala/ai/chronon/online/MetadataStore.scala:72]: can you add unit tests?

File: online/src/main/scala/ai/chronon/online/MetadataEndPoint.scala (hunk 1 above)
Comment (yuli-han) [reply online/src/main/scala/ai/chronon/online/MetadataEndPoint.scala:20]: @pengyu-hou Yep we have to. This is the data name we set on Mussel.

**Review by pengyu-hou — APPROVED** [review]

File: online/src/main/scala/ai/chronon/online/MetadataStore.scala [hunk 3]
```diff
@@ -68,6 +69,51 @@ class MetadataStore(kvStore: KVStore, val dataset: String = ChrononMetadataKey,
...
+              logger.error(
+                s"Failed to fetch conf for team $team at group_bys/$team, please check metadata upload to make sure the metadata has been uploaded")
+              throw e
+          }
+      },
+      { team => Metrics.Context(environment = "group_by.list.fetch", groupBy = team) }
+    )
+  }
+
+  lazy val getJoinByListByTeam: TTLCache[String, Try[Seq[String]]] = {
```
Comment (hzding621) [inline online/src/main/scala/ai/chronon/online/MetadataStore.scala:102]: nit: rename to `getJoinListByTeam`

**Review by hzding621 — COMMENTED** [review]
let's add some UT for this flow - we have `InMemoryKvStore` abstraction in UT

**Review by hzding621 — APPROVED** [review]

**Pull request merged**